pudl_scrapers/spiders/eia860m.py

```diff
@@ -0,0 +1,179 @@
+"""
+Spider for EIA 860 M.
+
+This module include the required infromation to establish a scrapy spider for
+EIA 860M. Most of the logic in here revolves around finding the right page,
+finding the right links from the page and storing how and where they should be
+saved. EIA 860 M is different from the other spiders in that we need the
+ability to specify both a month and a year to scrape. Right now the two modes
+are grab everything or grab a specific month and year combo.
+
+"""
+
+import logging
+from pathlib import Path
+import scrapy
+from scrapy.http import Request
+from time import strptime
+
+import pudl_scrapers
+
+
+logger = logging.getLogger(__name__)
+
+
+class Eia860MSpider(scrapy.Spider):
+    """Spider for Monthly EIA 860M."""
+
+    name = 'eia860m'
+    allowed_domains = ['www.eia.gov']
+
+    def __init__(self, year=None, month=None, *args, **kwargs):
+        """
+        Spider for scrapping EIA 860 Monthly.
+
+        If a year is specified a month must also be specified. Not all
+        year/month combos will provide data. The months available are quite
+        erradic. Most years all months are available, but in 2015 only July -
+        December are available and the most recent year's month should change
+        regularly. If a bad year/month combo is given nothing will be returned.
+        Check for availability before scrapping:
+        www.eia.gov/electricity/data/eia860m/
+
+
+
+        Args:
+            year (int): year of available EIA 860 M data. Currently only
+                available from 2015 through 2020. This works only in
+                conjunction with specifying a month. Default is None.
+            month (string): full name of month to grab. This works only in
+                conjunction with specifying a year. Default is None.
```

Comment:
I was imagining just a single parameter rather than two (since you have to have both) like `--yearmonth="2019-08"` which would then be consistent with the output filename (and could be automatically translated to the `August` string internally for interacting with the webpage).  But we'll usually just scrape the whole dataset all at once so maybe it's not even worth having the ability to split these out at all, especially since it's probably a very small dataset overall.